pinot-common/src/main/java/org/apache/pinot/common/utils/TlsUtils.java

```diff
@@ -301,48 +287,6 @@ private static final class SSLContextHolder {
         : SSL_CONTEXT_REF.get();
   }
 
-  /**
-   * Adapted from: https://svn.apache.org/viewvc/httpcomponents/oac
-   * .hc3x/trunk/src/contrib/org/apache/commons/httpclient/contrib/ssl/AuthSSLProtocolSocketFactory.java?view=markup
-   */
-  private static class PinotProtocolSocketFactory implements ProtocolSocketFactory {
```

Comment:
@apucher / @Jackie-Jiang, do we need this class? I am trying to identify the usage, any pointers?

Reply:
This is (was?) a helper to create and inject a custom SSL context into some of the network clients used in Pinot. If you don't find any remaining dependencies, this can be scrapped.

Reply:
imo, it's not getting used. All our test cases are green; however, there might also be a missing test case scenario.

We have [HttpClient](https://github.com/apache/pinot/blob/master/pinot-common/src/main/java/org/apache/pinot/common/utils/http/HttpClient.java#L100) class with `SSLConnectionSocketFactory` which is providing the same functionality based on `org.apache.httpcomponents`

Based on my understanding this is ola way of defining the custom HTTPs connection.